src/factory/client.ts

```diff
@@ -436,21 +438,34 @@ export const createClient = (
   network: Network,
   mnemonic: string,
   accountType: AccountType,
-  derivationPath: string
+  derivationPath: string,
+  publicKey?: string,
+  chainCode?: string
 ) => {
   const assetData = cryptoassets[asset];
```

Comment:
Can you do a check here that throws an error when publickey, chaincode is provided but client is not BTC?

Reply:
it works, tested with the wallet and evm accounts

Reply:
I'm sure it works, but my suggetion gives some feedback to a consumer that would call those parameters on a non BTC client, and expect it to work.